pkg/api/validation/validation.go

```diff
@@ -3471,6 +3469,29 @@ func validateFinalizerName(stringValue string, fldPath *field.Path) field.ErrorL
 	return field.ErrorList{}
 }
 
+// ValidateFinalizers tests if the finalizers name are valid, and if there are conflicting finalizers.
+func ValidateFinalizers(finalizers []string, fldPath *field.Path) field.ErrorList {
+	if fldPath == nil {
```

Comment:
fldPath already handles this when you do child, so you never have to check a fldPath for nil.